system_test/test_badge_server.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ast
+import os
+import requests
+import signal
+import subprocess
+
+from retrying import retry
+
+import unittest
+
+HOST_PORT = '0.0.0.0:8080'
+BASE_URL = 'http://0.0.0.0:8080/'
+PACKAGE_FOR_TEST = 'opencensus'
+
+RETRY_WAIT_PERIOD = 8000 # Wait 8 seconds between each retry
+RETRY_MAX_ATTEMPT = 10 # Retry 10 times
+
+EXPECTED_SVG = b'<svg xmlns="http://www.w3.org/2000/svg" xmlns:' \
```

Comment:
This looks scary... Should we put this in a separate file?

Reply:
Done.